src/pixelator/graph/graph.py

```diff
@@ -185,16 +187,31 @@ def layout_coordinates(
         :param only_keep_a_pixels: If true, only keep the a-pixels
         :param get_node_marker_matrix: Add a matrix of marker counts to each
                                        node if True.
+        :param cache: set to true in order to cache `cache_size` number of calls to
+                      this method. It will make subsequent calls to the layout method
+                      with the same settings much faster, at the cost of additional
+                      memory usage.
         :return: the coordinates and markers (if activated) as a dataframe
         :rtype: pd.DataFrame
         :raises: AssertionError if the provided `layout_algorithm` is not valid
         :raises: ValueError if the provided current graph instance is empty
         """
-        return self._backend.layout_coordinates(
-            layout_algorithm=layout_algorithm,
-            only_keep_a_pixels=only_keep_a_pixels,
-            get_node_marker_matrix=get_node_marker_matrix,
-        )
+        if cache:
+            return self._cached_layout_coordinates(
+                layout_algorithm=layout_algorithm,
+                only_keep_a_pixels=only_keep_a_pixels,
+                get_node_marker_matrix=get_node_marker_matrix,
+            )
+        else:
+            return self._backend.layout_coordinates(
+                layout_algorithm=layout_algorithm,
+                only_keep_a_pixels=only_keep_a_pixels,
+                get_node_marker_matrix=get_node_marker_matrix,
+            )
+
+    @lru_cache(maxsize=1)
```

Comment:
Since I don't want this to grow out of proportions, but e.g. plotting multiple markers on the same component is a rather common scenario but that will be massively sped up by only caching the layout computation.

Reply:
Understood this is mostly for iteration purposes when doing data analysis work.